--- a/target/linux/rockchip/image/armv7.mk
+++ b/target/linux/rockchip/image/armv7.mk
@@ -100,6 +100,15 @@ define Device/rockchip_rk3128_box
   DEVICE_TITLE := Rockchip RK3128 TV Box
   DEVICE_DTS := rk3128-tvbox-rk3128
   SUPPORTED_DEVICES += rk3128-tvbox
+  BLOCKSIZE := 1024k
+  PAGESIZE := 2048
+  IMAGE_SIZE := 15360k
+  IMAGES += factory.img sysupgrade.img
+  IMAGE/factory.img := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
+  IMAGE/sysupgrade.img := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
 endef
 TARGET_DEVICES += rockchip_rk3128_box

+define Device/tvbox_rk3128_generic
+  $(Device/rockchip_rk3128_box)
+  DEVICE_TITLE := Generic RK3128 TV Box (HDMI+2USB+LAN+WiFi)
+endef
+TARGET_DEVICES += tvbox_rk3128_generic
